sycl/test/devicelib/math_utils.hpp

```diff
@@ -1,23 +1,33 @@
 #ifndef MATH_UTILS
 #include <complex>
 #include <limits>
-using namespace std;
-// T must be float-point type
+
+// Since it is not proper to compare float point using operator ==, this
+// function measures whether the result of cmath function from kernel is
+// close to the reference and machine epsilon is used as threshold in this
+// function. T must be float-point type.
 template <typename T>
-bool is_about_FP(T x, T y) {
-  if (x == y)
-    return true;
+bool approx_equal_fp(T x, T y) {
+
+  // At least one input is nan
+  if (std::isnan(x) || std::isnan(y))
+    return std::isnan(x) && std::isnan(y);
+
+  // At least one input is inf
+  if (std::isinf(x) || std::isinf(y))
+    return (x == y);
+
+  // two finite
   else {
```

Comment:
Please, [don't use else after return](https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return).
Apply here and all the cases below.